src/storage/mvcc/write.rs

```diff
@@ -101,46 +103,74 @@ impl Write {
         }
     }
 
-    pub fn to_bytes(&self) -> Vec<u8> {
-        let mut b = Vec::with_capacity(1 + MAX_VAR_U64_LEN + SHORT_VALUE_MAX_LEN + 2);
-        b.push(self.write_type.to_u8());
-        b.encode_var_u64(self.start_ts).unwrap();
-        if let Some(ref v) = self.short_value {
-            b.push(SHORT_VALUE_PREFIX);
-            b.push(v.len() as u8);
-            b.extend_from_slice(v);
-        }
-        b
+    #[inline]
+    pub fn parse_type(mut b: &[u8]) -> Result<WriteType> {
+        let write_type_bytes = b.read_u8().map_err(|_| Error::BadFormatWrite)?;
+        WriteType::from_u8(write_type_bytes).ok_or(Error::BadFormatWrite)
     }
 
-    pub fn parse(mut b: &[u8]) -> Result<Write> {
-        if b.is_empty() {
-            return Err(Error::BadFormatWrite);
+    #[inline]
+    pub fn as_ref(&self) -> WriteRef<'_> {
```

Comment:
Could Write implement `Deref` then you wouldn't need `as_ref` all over

Reply:
And if not, can it be an AsRef impl?

Reply:
Both of them produce &T instead of U

Reply:
:-(